tests/components/yale/test_init.py

```diff
@@ -0,0 +1,238 @@
+        },
+    )
+    last_err = None
+    data = {ATTR_ENTITY_ID: "lock.a6697750d607098bae8d6baa11ef8063_name"}
+    try:
+        await hass.services.async_call(LOCK_DOMAIN, SERVICE_LOCK, data, blocking=True)
+    except HomeAssistantError as err:
+        last_err = err
+    assert str(last_err) == (
+        "A6697750D607098BAE8D6BAA11EF8063 Name: This should bubble up as its user"
+        " consumable"
+    )
+
+
+async def test_open_throws_hass_service_not_supported_error(
+    hass: HomeAssistant,
+) -> None:
+    """Test open throws correct error on entity does not support this service error."""
+    mocked_lock_detail = await _mock_operative_yale_lock_detail(hass)
+    await _create_yale_with_devices(hass, [mocked_lock_detail])
+    data = {ATTR_ENTITY_ID: "lock.a6697750d607098bae8d6baa11ef8063_name"}
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(LOCK_DOMAIN, SERVICE_OPEN, data, blocking=True)
+
+
+async def test_inoperative_locks_are_filtered_out(hass: HomeAssistant) -> None:
+    """Ensure inoperative locks do not get setup."""
+    yale_operative_lock = await _mock_operative_yale_lock_detail(hass)
+    yale_inoperative_lock = await _mock_inoperative_yale_lock_detail(hass)
+    await _create_yale_with_devices(hass, [yale_operative_lock, yale_inoperative_lock])
+
+    lock_abc_name = hass.states.get("lock.abc_name")
+    assert lock_abc_name is None
+    lock_a6697750d607098bae8d6baa11ef8063_name = hass.states.get(
+        "lock.a6697750d607098bae8d6baa11ef8063_name"
+    )
+    assert lock_a6697750d607098bae8d6baa11ef8063_name.state == STATE_LOCKED
+
+
+async def test_lock_has_doorsense(hass: HomeAssistant) -> None:
+    """Check to see if a lock has doorsense."""
+    doorsenselock = await _mock_doorsense_enabled_yale_lock_detail(hass)
+    nodoorsenselock = await _mock_doorsense_missing_yale_lock_detail(hass)
+    await _create_yale_with_devices(hass, [doorsenselock, nodoorsenselock])
+
+    binary_sensor_online_with_doorsense_name_open = hass.states.get(
+        "binary_sensor.online_with_doorsense_name_door"
+    )
+    assert binary_sensor_online_with_doorsense_name_open.state == STATE_ON
+    binary_sensor_missing_doorsense_id_name_open = hass.states.get(
+        "binary_sensor.missing_with_doorsense_name_door"
+    )
+    assert binary_sensor_missing_doorsense_id_name_open is None
+
+
+async def test_load_unload(hass: HomeAssistant) -> None:
+    """Config entry can be unloaded."""
+
+    yale_operative_lock = await _mock_operative_yale_lock_detail(hass)
+    yale_inoperative_lock = await _mock_inoperative_yale_lock_detail(hass)
+    config_entry, socketio = await _create_yale_with_devices(
+        hass, [yale_operative_lock, yale_inoperative_lock]
+    )
+
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
```

Comment:
Should we assert it's not loaded anymore?

Reply:
- [ ] yale - 
- [ ] august -